### PR TITLE
Luanti rename: builtin

### PR DESCRIPTION
**Pull request by Wuzzy2**
I've renamed "Minetest" to "Luanti", the new engine name, in builtin and settingtypes.

That's it.

### CODE REVIEW

**Comment by sfence** [comment]
For game texts which are also translated see #15309. 
There should be kept up to date also generated files that are connected to translation/gettext.

**Comment by Wuzzy2** [comment]
I think an update of the PO/POT files should be made separately, when the source files are done.

File: builtin/mainmenu/content/pkgmgr.lua [hunk 1]
```diff
@@ -840,7 +840,7 @@ function pkgmgr.get_contentdb_id(content)
 		return content.author:lower() .. "/" .. content.name
 	end
 
-	-- Until Minetest 5.8.0, Minetest Game was bundled with Minetest.
+	-- Until Luanti 5.8.0, Minetest Game was bundled with Luanti.
```
Comment (Zughy) [inline builtin/mainmenu/content/pkgmgr.lua:843]:
```suggestion
	-- Until version 5.8.0, Minetest Game was bundled with Luanti.
```

As it wasn't called Luanti back then

**Review by Zughy — COMMENTED** [review]
image

It'd make sense to also change these strings so to suggest `core.foo` and not `minetest.foo` (these aren't the only one)

**Comment by grorp** [comment]
Here's a PR for you: https://github.com/Wuzzy2/minetest/pull/4

**Comment by Wuzzy2** [comment]
I've done the proposed changes, except the line `minetest.exe` → `luanti.exe` proposed by @grorp.

I don’t know if this executable will be actually renamed.

**Comment by grorp** [comment]
> I don’t know if this executable will be actually renamed.

The rename happens in #15294.

**Review by grorp — APPROVED** [review]

**Comment by Zughy** [comment]
@Wuzzy2 rebase needed

File: builtin/common/serialize.lua [hunk 2]
```diff
@@ -204,18 +204,18 @@ local function dummy_func() end
...
 	if t ~= "string" then
-		error(("minetest.deserialize called with %s (expected string)."):format(t))
+		error(("core.deserialize called with %s (expected string)."):format(t))
 	end
 
 	local func, err = loadstring(str)
 	if not func then return nil, err end
 
-	-- math.huge was serialized to inf and NaNs to nan by Lua in Minetest 5.6, so we have to support this here
+	-- math.huge was serialized to inf and NaNs to nan by Lua in version 5.6, so we have to support this here
```
Comment (Desour) [inline builtin/common/serialize.lua:218]: Sounds like lua version 5.6.
```suggestion
	-- math.huge was serialized to inf and NaNs to nan by Lua in engine version 5.6, so we have to support this here
```

**Comment by Wuzzy2** [comment]
Got stuck in the rebase at this point:

```
builtin/mainmenu/credits.json
++<<<<<<< HEAD
 +              "Ælla Chiana Moskopp (erle) <erle@dieweltistgarnichtso.net> [Logo]",
++=======
+               "Nils Dagsson Moskopp (erlehmann) <nils@dieweltistgarnichtso.net> [Luanti logo]",
++>>>>>>> 38c905ad9 (Luanti rename: builtin)
```

These are the credits. I don't know which of these entries I have to keep. @grorp

**Comment by grorp** [comment]
"Ælla Chiana Moskopp (erle) <erle@dieweltistgarnichtso.net> [Logo]" is my version with the up-to-date name.

**Comment by Wuzzy2** [comment]
That doesn't answer my question. Again: Which of these two names do I keep? The first? The second? Both? Neither?

Because it feels weird to delete @erlehmann from the credits.

**Comment by grorp** [comment]
#15331

**Comment by grorp** [comment]
Still missing the minetest.exe change

**Comment by Wuzzy2** [comment]
Done.

File: builtin/client/chatcommands.lua [hunk 3]
```diff
@@ -1,4 +1,4 @@
--- Minetest: builtin/client/chatcommands.lua
+-- Luanti: builtin/client/chatcommands.lua
```
Comment (sfan5) [inline builtin/client/chatcommands.lua:1]: maybe we should just drop these? they seem pointless

**Review by sfan5 — APPROVED** [review]
LGTM otherwise

**Comment by Wuzzy2** [comment]
The headers have been removed. Yes, I agree, they're not important.

**Pull request merged**